design-proposals/flavors-and-preferences.md

```diff
@@ -0,0 +1,1023 @@
+  * [Memory](http://kubevirt.io/api-reference/main/definitions.html#_v1_memory)
+    * [guest](http://kubevirt.io/api-reference/main/definitions.html#_k8s_io_apimachinery_pkg_api_resource_quantity) 
+      * `VirtualMachineFlavorSpec.Memory.Guest`
+    * [hugepages](http://kubevirt.io/api-reference/main/definitions.html#_v1_hugepages)
+      * `VirtualMachineFlavorSpec.Memory.HugePages`
+
+## Types
+
+### `VirtualMachineFlavorSpec`
+
+```go
+// VirtualMachineFlavorSpec
+//
+// +k8s:openapi-gen=true1
+type VirtualMachineFlavorSpec struct {
+	CPU *CPUFlavor `json:"cpu"`
+
+	Memory *MemoryFlavor `json:"memory"`
+
+	// Optionally defines any GPU devices associated with the flavor.
+	//
+	// +optional
+	// +listType=atomic
+	GPUs []v1.GPU `json:"gpus,omitempty"`
+
+	// Optionally defines any HostDevices associated with the flavor.
+	//
+	// +optional
+	// +listType=atomic
+	HostDevices []v1.HostDevice `json:"hostDevices,omitempty"`
+
+	// Optionally defines the IOThreadsPolicy to be used by the flavor.
+	//
+	// +optional
+	IOThreadsPolicy *v1.IOThreadsPolicy `json:"ioThreadsPolicy,omitempty"`
+
+	// Optionally defines the LaunchSecurity to be used by the flavor.
+	//
+	// +optional
+	LaunchSecurity *v1.LaunchSecurity `json:"launchSecurity,omitempty"`
+}
+```
+
+### `CPUFlavor`
+
+```go
+// CPUFlavor
+//
+// +k8s:openapi-gen=true
+type CPUFlavor struct {
+
+	// Number of vCPUs to expose to the guest.
+	// The resulting CPU topology being derived from the optional PreferredCPUTopology attribute of CPUPreferences.
+	Guest uint32 `json:"guest"`
+
+	// Model specifies the CPU model inside the VMI.
+	// List of available models https://github.com/libvirt/libvirt/tree/master/src/cpu_map.
+	// It is possible to specify special cases like "host-passthrough" to get the same CPU as the node
+	// and "host-model" to get CPU closest to the node one.
+	// Defaults to host-model.
+	// +optional
+	Model string `json:"model,omitempty"`
+
+	// DedicatedCPUPlacement requests the scheduler to place the VirtualMachineInstance on a node
+	// with enough dedicated pCPUs and pin the vCPUs to it.
+	// +optional
+	DedicatedCPUPlacement bool `json:"dedicatedCPUPlacement,omitempty"`
+
+	// NUMA allows specifying settings for the guest NUMA topology
+	// +optional
+	NUMA *v1.NUMA `json:"numa,omitempty"`
+
+	// IsolateEmulatorThread requests one more dedicated pCPU to be allocated for the VMI to place
+	// the emulator thread on it.
+	// +optional
+	IsolateEmulatorThread bool `json:"isolateEmulatorThread,omitempty"`
+
+	// Realtime instructs the virt-launcher to tune the VMI for lower latency, optional for real time workloads
+	// +optional
+	Realtime *v1.Realtime `json:"realtime,omitempty"`
+}
+```
+
+### `MemoryFlavor`
+
+```go
+// FlavorMemory
+//
+// +k8s:openapi-gen=true
+type MemoryFlavor struct {
+
+	// Guest allows to specifying the amount of memory which is visible inside the Guest OS.
+	Guest *resource.Quantity `json:"guest,omitempty"`
```

Comment:
A Unit is appreciated here as well

Reply:
ACK, as above if it's fine to differ from the VM API I can do that here.